src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggingExtensions.cs

```diff
@@ -50,15 +50,40 @@ public static class OpenTelemetryLoggingExtensions
     /// <returns>The supplied <see cref="ILoggingBuilder"/> for call chaining.</returns>
     public static ILoggingBuilder AddOpenTelemetry(
         this ILoggingBuilder builder)
+        => AddOpenTelemetryInternal(builder, configureBuilder: null, configureOptions: null);
+
+    /// <summary>
+    /// Adds an OpenTelemetry logger named 'OpenTelemetry' to the <see cref="ILoggerFactory"/>.
+    /// </summary>
+    /// <remarks><inheritdoc cref="AddOpenTelemetry(ILoggingBuilder)" path="/remarks"/></remarks>
+    /// <param name="builder">The <see cref="ILoggingBuilder"/> to use.</param>
+    /// <param name="configure">Optional configuration action.</param>
+    /// <returns>The supplied <see cref="ILoggingBuilder"/> for call chaining.</returns>
+    public static ILoggingBuilder AddOpenTelemetry(
+        this ILoggingBuilder builder,
+        Action<OpenTelemetryLoggerOptions>? configure)
+        => AddOpenTelemetryInternal(builder, configureBuilder: null, configureOptions: configure);
+
+    private static ILoggingBuilder AddOpenTelemetryInternal(
+        ILoggingBuilder builder,
+        Action<LoggerProviderBuilder>? configureBuilder,
+        Action<OpenTelemetryLoggerOptions>? configureOptions)
     {
         Guard.ThrowIfNull(builder);
 
         builder.AddConfiguration();
 
+        var services = builder.Services;
+
         // Note: This will bind logger options element (eg "Logging:OpenTelemetry") to OpenTelemetryLoggerOptions
-        RegisterLoggerProviderOptions(builder.Services);
+        RegisterLoggerProviderOptions(services);
 
-        new LoggerProviderBuilderBase(builder.Services).ConfigureBuilder(
+        if (configureOptions != null)
```

Comment:
You have flipped the order for calling `services.Configure(configureOptions)` and `RegisterLoggerProviderOptions(services)`. Is this intentional?

Reply:
Ya I did this intentionally because I think it is a better logical order for things in the code:

* builder.AddConfiguration(); <- Sets up the services for logger provider configuration
* RegisterLoggerProviderOptions(services); <- Sets up the mapping of "Logging:OpenTelemetry" to `OpenTelemetryLoggerOptions`
* services.Configure(configureOptions); <- Registers the user-supplied `OpenTelemetryLoggerOptions` delegate

That being said, I didn't think it would change anything. I just did a couple tests and it does appear to be changing things.

Before if I had some config like this...

```json
  "Logging": {
    "LogLevel": {
      "Default": "Information"
    },
    "OpenTelemetry": {
      "IncludeFormattedMessage": true,
      "IncludeScopes": true,
      "ParseStateValues": true
    }
  },
```

...and code like this...

```csharp
appBuilder.Logging.AddOpenTelemetry(options =>
{
    options.IncludeFormattedMessage = false;
    options.AddConsoleExporter();
});
```

The order was...

* User callback fires. `IncludeFormattedMessage = false`
* Options bound. `IncludeFormattedMessage = true`

Final result: `IncludeFormattedMessage = true`

With this change the order is...

* Options bound. `IncludeFormattedMessage = true`
* User callback fires. `IncludeFormattedMessage = false`

Final result: `IncludeFormattedMessage = false`

Generally the approach we have taken is programmatic (code) wins over config so I would call the existing behavior a bug we should fix! But because this PR isn't logging related I'm going to revert the change here and then I'll maybe fix this on my next PR updating logging.

Reply:
Pushed the update. Nice catch by the way.